src/Components/Search/NewSearch/useTrackSearchResults.ts

```diff
@@ -0,0 +1,39 @@
+import { useUpdateEffect } from "@artsy/palette"
+import { useCallback, useState } from "react"
+import { useTracking } from "react-tracking"
+import * as DeprecatedSchema from "@artsy/cohesion/dist/DeprecatedSchema"
+import { ContextModule } from "@artsy/cohesion"
+import { shouldStartSearching } from "Components/Search/NewSearch/utils/shouldStartSearching"
+
+interface Options {
+  contextModule: ContextModule
+  searchQuery: string
+  hasSearchResults: boolean
+}
+
+export const useTrackSearchResults = (options: Options) => {
+  const tracking = useTracking()
+
+  // We use fetchCounter together with useUpdateEffect to track updating search query
+  const [fetchCounter, setFetchCounter] = useState(0)
+
+  const searchResultsUpdated = useCallback(() => {
+    setFetchCounter(prevCounter => prevCounter + 1)
+  }, [setFetchCounter])
```

Comment:
Why wouldn't you just call the trackEvent here, why the indirection with the counter state?

Reply:
* We should track `ActionType.SearchedAutosuggestWithResults` if there are results for the query
* We should track `ActionType.SearchedAutosuggestWithoutResults` if there are **no** results for the query
* Refetch `callback` returns only `error` (see attached screenshot), and we have no way to determine if there are results for the query or not. For this reason, `searchResultsUpdated` is called, which updates the `counter` and tracks [the event](https://github.com/artsy/force/blob/5f925aa83d2fbd4f1e043ac80b7b7b75fac072c6/src/Components/Search/NewSearch/useTrackSearchResults.ts#L28-L35) with the current results for the query

<img width="736" alt="for-damon" src="https://github.com/artsy/force/assets/3513494/18288492-53cd-415f-8c91-a05f4d115215">

Reply:
Tried [this approach](https://github.com/artsy/force/commit/a85466e2afd760ebecce1a746eef922e46c29d36). But unfortunately it does **not** work correctly (events are duplicated)

https://github.com/artsy/force/assets/3513494/853bdfb9-591a-449b-88c9-73ad48951c7b